File: Android/Reddit/app/src/main/java/com/ognjenlazic/reddit/network/service/ReactionService.kt
package com.ognjenlazic.reddit.network.service

import com.ognjenlazic.reddit.model.Karma
import com.ognjenlazic.reddit.model.Reaction
import com.ognjenlazic.reddit.model.User
import com.ognjenlazic.reddit.network.model.ReactionResponse
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface ReactionService {
    @GET("reaction/post-karma/{id}")
    suspend fun getPostKarma(@Path("id") id: Int): Karma

    @GET("reaction/user-karma/{userId}")
    suspend fun getUserKarma(@Path("userId") id: Long): Karma

    @GET("reaction/comment-karma/{commentId}")
    suspend fun getCommentKarma(@Path("commentId") id: Int): Karma

    @GET("reaction/post/{postId}")
    suspend fun getAllPostReactions(@Path("postId") postId: Int): List<ReactionResponse>

    @GET("reaction/comment/{commentId}")
    suspend fun getAllCommentReactions(@Path("commentId") commentId: Int): List<ReactionResponse>

    @POST("reaction")
    suspend fun createReaction(@Body reaction: Reaction)
}

File: Frontend/reddit/src/app/services/reaction_service/reaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClientModule, HttpParams, HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Post } from 'src/app/model/post';
import { Comment } from 'src/app/model/comment';
import { Karma } from 'src/app/helper/model/karma';

@Injectable({
  providedIn: 'root'
})
export class ReactionService {
  private headers = new HttpHeaders({ 'Content-Type': 'application/json' });
  private readonly baseURL = "/api/api/reaction/"
  constructor(
    private http: HttpClient
  ) { }

  createReaction(reaction: string, post: Post, userId: number): Observable<any> {
    return this.http.post(this.baseURL,
      {
        reactionId: 10,
        reactionType: reaction,
        userId: userId,
        timestamp: new Date(),
        postId: post.id,
        commentId: null,
      },
      { headers: this.headers, responseType: 'json' });
  }

  getAllPostReactions(postId: number) : Observable<any>{
    let queryParams = {};
    queryParams = {
      headers: this.headers,
      observe: 'response'
    };
    return this.http.get(this.baseURL + "post/" + postId, queryParams);
  }

  getAllCommentReacrions(postId: number): Observable<any> {
    let queryParams = {};
    queryParams = {
      headers: this.headers,
      observe: 'response'
    };
    return this.http.get(this.baseURL + "comment/" + postId, queryParams);
  }

  createCommentReaction(reaction: string, comment: Comment, userId: number): Observable<any> {
    return this.http.post(this.baseURL,
      {
        reactionId: 10,
        reactionType: reaction,
        userId: userId,
        timestamp: new Date(),
        postId: null,
        commentId: comment.commentId,
      },
      { headers: this.headers, responseType: 'json' });
  }

  getPostKarma(postId: number): Observable<any> {
    let queryParams = {};
    queryParams = {
      headers: this.headers,
      observe: 'response'
    };
    return this.http.get(this.baseURL + "post-karma/" + postId, queryParams);

  }

  getUserKarma(userId: number): Observable<any> {
    let queryParams = {};
    queryParams = {
      headers: this.headers,
      observe: 'response'
    };
    return this.http.get(this.baseURL + "user-karma/" + userId, queryParams);

  }

  getCommentKarma(commentId: number): Observable<any> {

    let queryParams = {};
    queryParams = {
      headers: this.headers,
      observe: 'response'
    };
    return this.http.get(this.baseURL + "comment-karma/" + commentId, queryParams);
    //  this.http.get<Karma>(this.baseURL + "comment-karma/" + commentId);
  }

}
